### PR TITLE
TR-1286: Basic editing for Prototype fullscreen code editor with preview

### PR DESCRIPTION
**Pull request by bkemper**
This is a subtask [TR-1286](https://jira.int.messagesystems.com/browse/TR-1286) of [TR-1204](https://jira.int.messagesystems.com/browse/TR-1204).

### What Changed
- Set title to template name
- Added tabs (can deep link to a tab)
- Dark mode editor with save as draft button

Screen Shot 2019-05-17 at 12 24 08 PM

### How To Test
- Visit http://app.sparkpost.test/templates/edit/amp/next
- Click a tab (should update url)
- Modify content
- Click "Save As Draft" (should disable button while processing)
- Refresh page (should have saved most recently saved changes)

### CODE REVIEW

**Comment by bkemper** [comment]
@rajumsys This is ready for review.

File: src/pages/templates/components/EditPrimaryArea.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Button } from '@sparkpost/matchbox';
+import useEditorContext from '../hooks/useEditorContext';
+
+const EditPrimaryArea = () => {
+  const { content, draft, isDraftUpdating, updateDraft } = useEditorContext();
```
Comment (rajumsys) [inline src/pages/templates/components/EditPrimaryArea.js]: can't follow where these (like `updateDraft`) coming from

File: src/pages/templates/context/EditorContext.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  children,
+  value: { getDraft, getPublished, ...value }
+}) => {
+  const contentState = useEditorContent(value.draft);
+  const { requestParams } = useRouter();
+
+  useEffect(() => {
+    getDraft(requestParams.id, requestParams.subaccount);
+    getPublished(requestParams.id, requestParams.subaccount);
+  }, [getDraft, getPublished, requestParams.id, requestParams.subaccount]);
```
Comment (rajumsys) [inline src/pages/templates/context/EditorContext.js]: are functions in deps list helpful?

Comment (bkemper) [reply src/pages/templates/context/EditorContext.js]: The `react-hooks/exhaustive-deps` eslint rule requires and will auto set them.

File: src/pages/templates/components/EditPrimaryArea.js (hunk 1 above)
Comment (bkemper) [reply src/pages/templates/components/EditPrimaryArea.js]: The `EditAndPreviewPage` container provides `props` to `EditorContextProvider` as `value` prop.  The `useEditorContext` hook provides that value prop as the return value.

File: src/pages/templates/context/EditorContext.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useEffect } from 'react';
+import useRouter from 'src/hooks/useRouter';
+import useEditorContent from '../hooks/useEditorContent';
+
+const EditorContext = createContext();
+
+export const EditorContextProvider = ({
+  children,
+  value: { getDraft, getPublished, ...value }
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/pages/templates/context/EditorContext.js]: The `value` prop is provided by the container.

**Comment by bkemper** [comment]
> Can we make container the top level module, and then move the page inside a directory? basically swapping place for page & container!

I agree it is not ideal.  I believe the containers got split out to be excluded from test coverage.  To make another page directory might look a little funny, 'pages/templates/pages'.  How about we just dissolve the container director and move them to the main directory to live alongside page components?

**Comment by rajumsys** [comment]
> How about we just dissolve the container director and move them to the main directory to live alongside page components?

Yes, that's fine too.

**Comment by bkemper** [comment]
Rebased.

**Pull request merged**